docs/architecture/maui/dependency-injection.md

```diff
@@ -1,9 +1,192 @@
+
+By specifying dependencies as interface types, dependency injection enables decoupling of the concrete types from the code that depends on these types. It generally uses a container that holds a list of registrations and mappings between interfaces and abstract types, and the concrete types that implement or extend these types.
+
+There are also other types of dependency injection, such as *property setter injection*, and *method call injection*, but they are less commonly seen. Therefore, this chapter will focus solely on performing constructor injection with a dependency injection container.
+
+## Introduction to dependency injection
+
+Dependency injection is a specialized version of the Inversion of Control (IoC) pattern, where the concern being inverted is the process of obtaining the required dependency. With dependency injection, another class is responsible for injecting dependencies into an object at runtime. The following code example shows how the `ProfileViewModel` class is structured when using dependency injection:
+
+```csharp
+private readonly ISettingsService _settingsService;
+private readonly IAppEnvironmentService _appEnvironmentService;
+
+public ProfileViewModel(
+    IAppEnvironmentService appEnvironmentService,
+    IDialogService dialogService, INavigationService navigationService, ISettingsService settingsService)
+    : base(dialogService, navigationService, settingsService)
+{
+    _appEnvironmentService = appEnvironmentService;
+    _settingsService = settingsService;
+
+    // Omitted for brevity
+}
+```
+
+The `ProfileViewModel` constructor receives multiple interface object instances as an arguments, injected by another class. The only dependency in the `ProfileViewModel` class is on the interface types. Therefore, the `ProfileViewModel` class doesn't have any knowledge of the class that's responsible for instantiating the interface objects. The class that's responsible for instantiating the interface objects, and inserting it into the `ProfileViewModel` class, is known as the *dependency injection container*.
+
+Dependency injection containers reduce the coupling between objects by providing a facility to instantiate class instances and manage their lifetime based on the configuration of the container. During the objects creation, the container injects any dependencies that the object requires into it. If those dependencies have not yet been created, the container creates and resolves their dependencies first.
+
+> [!NOTE]
+> Dependency injection can also be implemented manually using factories. However, using a container provides additional capabilities such as lifetime management, and registration through assembly scanning. It is recommended to dependency injection using a container for .NET MAUI applications.
+
```

Comment:
```suggestion
```

I'd probably omit this, as you're not showing how to do it manually anyway. It's just a bit unnecessary.

Reply:
Makes sense to me. I was just thinking it _could_ be helpful, but in the grand scheme, it is likely more confusing and unnecessary.